image/helpers.go

```diff
@@ -108,16 +109,28 @@ func StoreAssertionFetcher(sto Store, dlOpts *DownloadOptions, db *asserts.Datab
 }
 
 // FetchAndCheckSnapAssertions fetches and cross checks the snap assertions matching the given snap file using the provided asserts.Fetcher and assertion database.
-func FetchAndCheckSnapAssertions(snapPath string, info *snap.Info, f asserts.Fetcher, db asserts.RODatabase) error {
+func FetchAndCheckSnapAssertions(snapPath string, info *snap.Info, f asserts.Fetcher, db asserts.RODatabase) (*asserts.SnapDeclaration, error) {
 	sha3_384, size, err := asserts.SnapFileSHA3_384(snapPath)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
+	// this assumes series "16"
 	if err := snapasserts.FetchSnapAssertions(f, sha3_384); err != nil {
-		return fmt.Errorf("cannot fetch snap signatures/assertions: %v", err)
+		return nil, fmt.Errorf("cannot fetch snap signatures/assertions: %v", err)
 	}
 
 	// cross checks
-	return snapasserts.CrossCheck(info.Name(), sha3_384, size, &info.SideInfo, db)
+	if err := snapasserts.CrossCheck(info.Name(), sha3_384, size, &info.SideInfo, db); err != nil {
+		return nil, err
+	}
+
+	a, err := db.Find(asserts.SnapDeclarationType, map[string]string{
+		"series":  release.Series,
+		"snap-id": info.SnapID,
+	})
```

Comment:
yes, we just get hold of the cross-verified snap-declaration and return it (CrossCheck is used also in snapd itself and changing its signature doesn't make as much sense), anyway the style is common also inside snapd,  a main task verify the assertions, later tasks get hold of the snap-declaration (from the db) later to do their work (this is just done very close together)